crates/iroha_core/src/smartcontracts/wasm.rs

```diff
@@ -957,6 +960,7 @@ impl<'wrld, 'block: 'wrld, 'state: 'block> Runtime<state::SmartContract<'wrld, '
     fn get_smart_contract_context(state: &state::SmartContract) -> payloads::SmartContractContext {
         payloads::SmartContractContext {
             authority: state.authority.clone(),
+            curr_block: state.state.0.curr_block,
```

Comment:
> 5. smart contract and trigger execution (as latest current block header)

There seems to be an error on documentation side:
Since the execution phase of a smartcontract (wasm as a transaction executable) is different from that of a trigger, each pair (block height, what executions refer the block header) is as follows:

|                                                           | >>>>>>>>>>     | block committed | >>>>>>>>>> | triggers executed |
| --------------------------------------------------------- | -------------- | --------------- | ---------- | ----------------- |
| current: `StateTransaction.curr_block`                    | k, transaction |                 | k, trigger |                   |
| latest committed: `CommonState.specific_state.curr_block` | k-1, query     |                 | k, query   |                   |

So a smartcontract refers the current (validating) block header, not the latest one. On the other hand, it is true that a block header that a trigger refers is current and latest committed

Reply:
You are correct. But this is only because our trigger execution model is broken

Reply:
Yes, when transaction and trigger execution are integrated and sequenced, such a discussion would be unnecessary

Reply:
so this should be fixed for queries that are executing as part of a transaction (like from inside a trigger). They should receive current block header instead of latest block header. Do you agree?

Any other issues to address?

Reply:
opened #5171